File: task_equivalents/llm_translations/llama/base/python/machine_learning_workflows/01_pipeline_cv_gridsearch_classification.py
import pandas as pd
from sklearn import svm
from sklearn.model_selection import GridSearchCV
from sklearn.preprocessing import StandardScaler
from sklearn.compose import ColumnTransformer
from sklearn.pipeline import Pipeline

# Load data
Auto = pd.read_csv('data/csv/Auto.csv', na_values='?').dropna().set_index('name')

# Create new column
Auto['mpg01'] = (Auto['mpg'] > Auto['mpg'].median()).astype(int)

# Select relevant columns
var_no_scale = ['cylinders', 'year', 'origin', 'mpg01']
var_to_scale = ['displacement', 'horsepower', 'weight', 'acceleration']
Auto = Auto[var_to_scale + var_no_scale]

# Scale relevant columns
scaler = ColumnTransformer([('scaler', StandardScaler(), var_to_scale)], remainder='passthrough')
x = Auto.drop('mpg01', axis=1)
y = Auto['mpg01']

# Define tuning parameters
ctry = [0.01, 0.1, 1, 10, 100, 1000]
param_grid_linear = {'svm__C': ctry}
param_grid_poly = {'svm__C': ctry, 'svm__degree': [2, 3, 4, 5]}
param_grid_radial = {'svm__C': ctry, 'svm__gamma': [0.5, 1, 2, 3]}

# Define pipeline and tune
pipeline = Pipeline([('scaler', scaler), ('svm', svm.SVC())])

# Linear kernel
grid_search_linear = GridSearchCV(pipeline, param_grid_linear, cv=5)
grid_search_linear.fit(x, y)
print("Linear Kernel:")
print(grid_search_linear.best_params_)
print(grid_search_linear.best_score_)

# Polynomial kernel
param_grid_poly = {'svm__C': ctry, 'svm__degree': [2, 3, 4, 5], 'svm__kernel': ['poly']}
pipeline_poly = Pipeline([('scaler', scaler), ('svm', svm.SVC(kernel='poly'))])
grid_search_poly = GridSearchCV(pipeline_poly, {'svm__C': ctry, 'svm__degree': [2, 3, 4, 5]}, cv=5)
grid_search_poly.fit(x, y)
print("\nPolynomial Kernel:")
print(grid_search_poly.best_params_)
print(grid_search_poly.best_score_)

# Radial kernel
pipeline_radial = Pipeline([('scaler', scaler), ('svm', svm.SVC(kernel='rbf'))])
grid_search_radial = GridSearchCV(pipeline_radial, param_grid_radial, cv=5)
grid_search_radial.fit(x, y)
print("\nRadial Kernel:")
print(grid_search_radial.best_params_)
print(grid_search_radial.best_score_)

File: task_equivalents/llm_translations/llama/base/r/machine_learning_workflows/01_pipeline_cv_gridsearch_classification.R
# Install and load necessary libraries
# install.packages(c("readr", "dplyr", "caret", "kernlab"))
library(readr)
library(dplyr)
library(caret)
library(kernlab)

# Load data
Auto <- read_csv('data/csv/Auto.csv', na = c("?")) %>%
  drop_na()

# Create binary variable mpg01
Auto$mpg01 <- ifelse(Auto$mpg > median(Auto$mpg, na.rm = TRUE), 1, 0)
Auto <- Auto %>%
  select(-c(mpg, name))

# Define variables to scale and not to scale
var_no_scale <- c('cylinders', 'year', 'origin', 'mpg01')
var_to_scale <- c('displacement', 'horsepower', 'weight', 'acceleration')

# Scale variables
scaled_var <- scale(Auto[, var_to_scale])
df <- data.frame(scaled_var, Auto[, var_no_scale])

# Split data into features and target
x <- df[, !names(df) %in% "mpg01"]
y <- df$mpg01

# Define train control
train_control <- trainControl(method = "cv", number = 5, 
                               savePredictions = "final", 
                               classProbs = TRUE)

# Linear kernel
tune_grid_linear <- expand.grid(C = c(0.01, 0.1, 1, 10, 100, 1000))
set.seed(123) # For reproducibility
linear_svm <- train(x, y, 
                    method = "svmLinear", 
                    tuneGrid = tune_grid_linear, 
                    trControl = train_control)

# Polynomial kernel
tune_grid_poly <- expand.grid(C = c(0.01, 0.1, 1, 10, 100, 1000), 
                              degree = c(2, 3, 4, 5), 
                              scale = c(1))
set.seed(123) # For reproducibility
poly_svm <- train(x, y, 
                  method = "svmPoly", 
                  tuneGrid = tune_grid_poly, 
                  trControl = train_control)

# Radial kernel
tune_grid_rbf <- expand.grid(sigma = c(0.5, 1, 2, 3), 
                             C = c(0.01, 0.1, 1, 10, 100, 1000))
# Map C and sigma to the required format for kernlab's ksvm
tune_grid_rbf <- expand.grid(C = c(0.01, 0.1, 1, 10, 100, 1000), 
                             sigma = c(0.5, 1, 2, 3))
set.seed(123) # For reproducibility
rbf_svm <- train(x, y, 
                 method = "svmRadial", 
                 tuneGrid = tune_grid_rbf, 
                 trControl = train_control)